hack/run-hive-locally.sh

```diff
@@ -0,0 +1,296 @@
+    fi
+
+    # Restore Hive
+    case "${component}" in
+    "$hive_component_operator")
+        hl_echo "Scaling up Deployment/hive-operator"
+        oc scale -n "$hive_operator_ns" deploy/hive-operator --replicas="$hive_operator_replicas"
+        ;;
+    "$hive_component_controllers")
+        kill_background_jobs
+        hl_echo "Disabling maintenance mode"
+        oc patch hiveconfig hive -p '{"spec": {"maintenanceMode": false}}' --type merge
+        ;;
+    *)
+        hl_echo "Invalid component $component" >&2
+        ;;
+    esac
+
+    # Clean up artifacts
+    if [[ -d "./$artifacts_folder" ]] && [[ "$cleanup_artifacts" = "true" ]]; then
+        hl_echo "Cleaning up artifacts"
+        rm -r "$artifacts_folder"
+    fi
+
+    # Propagate the exit signal to stop script execution
+    exit 0
+}
+
+# Configuration
+hive_component_operator="hive-operator"
+hive_component_controllers="hive-controllers"
+hive_components_to_choose_from=("$hive_component_operator" "$hive_component_controllers")
+hive_operator_ns="hive"
+hive_controllers_ns="$(oc get hiveconfig hive -o jsonpath='{.spec.targetNamespace}')"
+hive_controllers_ns="${hive_controllers_log_level:-hive}"
+hive_operator_replicas="$(oc get deploy "$hive_component_operator" -n "$hive_operator_ns" -o jsonpath='{.spec.replicas}')"
+hive_operator_log_level="debug"
+hive_controllers_log_level="debug"
+artifacts_folder="artifacts"
+cleanup_artifacts="false"
+hc_spec_generation=0
+kill_job_grace_period_seconds=3
+
+# Parse long options
+while [ $# -gt 0 ]; do
+    case "$1" in
+    "--hive-operator-ns="*)
+        hive_operator_ns="${1#*=}"
+        ;;
+    "--hive_operator_log_level="*)
+        hive_operator_log_level="${1#*=}"
+        ;;
+    "--hive_controllers_log_level="*)
+        hive_controllers_log_level="${1#*=}"
+        ;;
+    "--artifacts_folder="*)
+        artifacts_folder="${1#*=}"
+        ;;
+    "--cleanup_artifacts")
+        cleanup_artifacts="true"
+        ;;
+    "--help")
+        usage
+        exit 0
+        ;;
+    *)
+        hl_echo "Invalid option: $1" >&2
+        usage
+        exit 1
+        ;;
+    esac
+    shift
+done
+
+# Check if artifacts folder exists. Create it if not.
+if [[ -d "./$artifacts_folder" ]]; then
+    hl_echo "Artifacts folder $artifacts_folder exists and contains the following files:"
+    ls "./$artifacts_folder"
+
+    read -r -p "Clear folder and proceed (y/n, default=n) ? " clear_folder
+    case "${clear_folder:-n}" in
+    y)
+        rm ./"$artifacts_folder"/* || true
+        ;;
+    n)
+        hl_echo "Exiting"
+        exit 0
+        ;;
+    *)
+        hl_echo "Invalid choice, exiting"
+        exit 1
+        ;;
+    esac
+else
+    hl_echo "Creating artifacts folder $artifacts_folder"
+    mkdir "$artifacts_folder"
+fi
+
+# Set trap
+trap restore_and_cleanup EXIT SIGINT
+
+# Run the selected component locally
+PS3="Which component to run locally ? "
+select component in "${hive_components_to_choose_from[@]}"; do
```

Comment:
Again, I believe there should be a way to be non-interactive. Why not a mandatory param to indicate which component we care about?